spec/ios_version_helper_spec.rb

```diff
@@ -0,0 +1,71 @@
+require 'spec_helper'
+require_relative '../lib/fastlane/plugin/wpmreleasetoolkit/helper/ios/ios_version_helper'
+require_relative '../lib/fastlane/plugin/wpmreleasetoolkit/models/app_version'
+require_relative '../lib/fastlane/plugin/wpmreleasetoolkit/models/build_code'
+
+describe Fastlane::Helper::Ios::VersionHelper do
+  describe 'write the provided key and value to an xcconfig file' do
+    context 'when the .xcconfig file exists' do
+      it 'writes the key-value pair to the .xcconfig file' do
+        file_path = 'spec/test-data/test.xcconfig'
+        key = 'BUILD_NUMBER'
+        value = '1234'
+        described_class.write_to_xcconfig_file(key, value, file_path)
```

Comment:
Wouldn't this modify the `spec/test-data` spec file when you run the tests?

I'd instead suggest to copy the fixture file in a temp dir, then run the test and expectations on that file in the temp dir.

You can use `in_tmp_dir` and friend helper methods from the `spec_helper` for that. In any case, you shouldn't run on the test fixture, as that fixture isn't supposed to be modified by running the tests (and otherwise it could change the behavior of subsequent tests that expect to have a specific value and fixed content in that file)

Tbh I'm pretty surprised that `Xcodeproj` wouldn't work with `with_tmp_file` like for Android (as you mentioned in the comment above)… can you give more details as to what doesn't work and which error you get exactly when you tried it? Was it a crash in `Xcodeproj`, or just that it didn't do anything, or something else?

Reply:
@AliSoftware Okay, I did get the `with_tmp_file` working with `Xcodeproj`. Here's the commit with those updates: https://github.com/wordpress-mobile/release-toolkit/commit/7bef03bcaa0b7387c93c080ceff92a0b4ab72f95#diff-f2e80d7c2d207cd09dc7b9a9e9989b2b33d85ba979c936001f4d3e505bdc72fcR63

That link goes to this line in `ios_version_file.rb`: `File.write(xcconfig_path, config.to_s)`. That works, but doesn't use the `save_as` method from `Xcodeproj`. When I tried to use `save_as`, as in `config.save_as(file_path)`, it gave me this error. 

```
  1) Fastlane::Wpmreleasetoolkit::Versioning::IOSVersionFile write the provided version numbers to an xcconfig file when the .xcconfig file exists writes the versions to the .xcconfig file
     Failure/Error: config.save_as(xcconfig_path)

     NoMethodError:
       private method `open' called for #<String:0x0000000106e78978>
     # ./vendor/bundle/ruby/2.7.0/gems/xcodeproj-1.22.0/lib/xcodeproj/config.rb:110:in `save_as'
     # ./lib/fastlane/plugin/wpmreleasetoolkit/versioning/files/ios_version_file.rb:39:in `write'
     # ./spec/ios_version_file_spec.rb:27:in `block (5 levels) in <top (required)>'
     # ./spec/spec_helper.rb:96:in `block in with_tmp_file'
     # ./spec/spec_helper.rb:84:in `block (2 levels) in in_tmp_dir'
     # ./spec/spec_helper.rb:83:in `chdir'
     # ./spec/spec_helper.rb:83:in `block in in_tmp_dir'
     # ./spec/spec_helper.rb:82:in `in_tmp_dir'
     # ./spec/spec_helper.rb:91:in `with_tmp_file'
     # ./spec/ios_version_file_spec.rb:26:in `block (4 levels) in <top (required)>'
     # ./vendor/bundle/ruby/2.7.0/gems/buildkite-test_collector-2.3.1/lib/buildkite/test_collector/library_hooks/rspec.rb:22:in `block (2 levels) in <top (required)>'
     # ./vendor/bundle/ruby/2.7.0/gems/webmock-3.18.1/lib/webmock/rspec.rb:37:in `block (2 levels) in <top (required)>'
```

When I was investigating that, it sounds like the error could be due to trying to write an object like `config` directly to a file? But I'm not sure if that's correct. If it is correct, any ideas about how `Xcodeproj.save_as` could be used to save the updated config to the file? I thought I was using it in the right way, but it feels like I'm missing something. 

Either way, `File.write` does work correctly, but I'd love to use the `Xcodeproj` method if it's intended to be used like this.

Reply:
Oh I think that's because apparently, [`Xcodeproj::Config#save_as` expects its parameter to be a `Pathname`, not a `String`](https://github.com/CocoaPods/Xcodeproj/blob/master/lib/xcodeproj/config.rb#L100-L105).

So I think if you want to use `save_as` you can use `Pathname.new(file_path)` instead of passing `file_path` as a `String` directly (tbh I'd have hoped that the implementation of `save_as` would do that for us, but alas 🙃 )
That being said, using `File.write(file_path, config.to_s)` works too 🙃

Reply:
Yep, `Pathname.new(file_path)` was the key. Thanks! I'll submit a change to `Xcodeproj` after this project is done to improve that behavior.

Reply:
Ping me on it when you do, I'm a core team member over there so I should be able to review and merge it when it's ready 🙂